src/repr/src/row/encoding2.rs

```diff
@@ -615,10 +615,10 @@ impl DatumColumnEncoder {
                 let offsets = OffsetBuffer::<i32>::from_lengths(lengths.iter().copied());
                 let nulls = nulls.as_mut().map(|n| NullBuffer::from(n.finish()));
 
-                let key_field =
-                    Arc::new(Field::new("key", keys.data_type().clone(), nulls.is_some()));
-                let val_field =
-                    Arc::new(Field::new("val", vals.data_type().clone(), nulls.is_some()));
+                // The keys and values in a Map can never be null, instead a
+                // field being null is maintained by the higher level MapArray.
+                let key_field = Arc::new(Field::new("key", keys.data_type().clone(), false));
+                let val_field = Arc::new(Field::new("val", vals.data_type().clone(), false));
```

Comment:
this is unrelated to the issue of empty `RelationDesc`s, but was found with the new proptests.